platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ImportMigrationDirectoryEntryImpl.java

```diff
@@ -0,0 +1,195 @@
+
+    this.lastModified = (modified != null) ? modified : -1;
+  }
+
+  @Override
+  public long getLastModifiedTime() {
+    return lastModified;
+  }
+
+  @Override
+  public boolean restore(boolean required) {
+    if (restored == null) {
+      super.restored = false; // until proven otherwise
+      if (LOGGER.isDebugEnabled()) {
+        LOGGER.debug("Importing {}...", toDebugString());
+      }
+      // a directory is always exported by the framework, as such we can safely extend our
+      // privileges
+      AccessUtils.doPrivileged(
+          () -> {
+            final PathUtils pathUtils = getContext().getPathUtils();
+            final Path apath = getAbsolutePath();
+            // find all existing files ans keep track of it relative from ddf.home to absolute path
+            final Map<Path, Path> existingFiles =
+                FileUtils.listFiles(
+                        apath.toFile(), TrueFileFilter.INSTANCE, TrueFileFilter.INSTANCE)
+                    .stream()
+                    .map(File::toPath)
+                    .collect(Collectors.toMap(pathUtils::relativizeFromDDFHome, p -> p));
+
+            // it is safe to ignore the 'required' parameter since if we get here, we have a
+            // directory
+            // exported to start with and all files underneath are always optional so pass false to
+            // restore()
+            if (fileEntries
+                .stream()
+                .peek(me -> existingFiles.remove(me.getPath()))
+                .map(me -> me.restore(false))
+                .reduce(true, Boolean::logicalAnd)) {
+              if (!filtered) {
+                // all files from the original system were exported under this directory, so remove
+                // all files that were not on the original system but are found on the current one
+                final MigrationReport report = getReport();
+
+                existingFiles.forEach((p, ap) -> PathUtils.cleanQuietly(ap, report));
+                // cleanup all empty directories left underneath this entry's path
+                PathUtils.cleanQuietly(apath, report);
+              }
+              SecurityLogger.audit("Imported directory {}", apath);
+              super.restored = true;
+            } else {
+              SecurityLogger.audit("Error importing directory {}", apath);
+              getReport()
+                  .record(
+                      new MigrationException(
+                          Messages.IMPORT_PATH_COPY_ERROR,
+                          getPath(),
+                          pathUtils.getDDFHome(),
+                          "some directory entries failed"));
+            }
+          });
+    }
+    return restored;
+  }
+
+  @Override
+  public boolean restore(boolean required, PathMatcher filter) {
+    Validate.notNull(filter, "invalid null path filter");
+    if (restored == null) {
+      super.restored = false; // until proven otherwise
```

Comment:
I don't think you need isDebugEnabled() here when using slf4j. https://www.slf4j.org/faq.html#logging_performance

Reply:
Unfortunately, static analysis will complain about the call to debugString(). I wouldn't have put it otherwise :-(